ipatests/test_ipaserver/test_install/test_service.py

```diff
@@ -36,3 +42,64 @@ def test_format_seconds():
     assert service.format_seconds(62) == '1 minute 2 seconds'
     assert service.format_seconds(120) == '2 minutes'
     assert service.format_seconds(125) == '2 minutes 5 seconds'
+
+
+def check_dir(dir, username):
+    user_pw = pwd.getpwnam(username)
+    stats = os.stat(dir)
+    mode = stats.st_mode
+    if not stat.S_ISDIR(mode):
+        return False
+
+    # path should be owned by either root or our user
+    if stats.st_uid and not stats.st_uid == user_pw.pw_uid:
+        return False
+
+    if stats.st_uid == user_pw.pw_uid:
+        return mode & stat.S_IXUSR
+
+    # there is no check for suppl groups
+    if stats.st_gid == user_pw.pw_gid:
+        return mode & stat.S_IXGRP
+
+    return mode & stat.S_IXOTH
+
+
+def check_owned_path(path, username):
+    parts = pathlib.Path(path).parts
+    for dir in parts:
+        assert check_dir(os.path.sep.join(parts[:parts.index(dir) + 1]),
+                         username)
+
+
+def test_service_keytab_permissions():
+    # HTTP service keytab
+    username = constants.GSSPROXY_USER
+    assert username
+    http_keytab = paths.HTTP_KEYTAB
+    assert http_keytab
+    user_pw = pwd.getpwnam(username)
+
+    stats = os.stat(http_keytab)
+    assert stats.st_uid == user_pw.pw_uid
+    assert stats.st_gid == user_pw.pw_gid
+
+    mode = stats.st_mode
+    assert stat.S_ISREG(mode)
+    assert stat.S_IMODE(mode) == 0o600
+
+    # a parent dir should be writable by our user
+    parent_dir = pathlib.Path(http_keytab).parent
```

Comment:
Do you mean the parent directory ``/var/lib/ipa`` or the directory ``/var/lib/ipa/gssproxy`` that contains the file ``/var/lib/ipa/gssproxy/http.keytab``?

pathlib is not available in Python 2.7. This would make a backport to 4.7 branch harder. ``os.path.dirname('/var/lib/ipa/gssproxy/http.keytab')`` returns the directory ``/var/lib/ipa/gssproxy``.